addons/base_iban/static/src/components/iban_widget/iban_widget.js

```diff
@@ -0,0 +1,45 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { CharField } from "@web/views/fields/char/char_field";
+import { useDebounced } from "@web/core/utils/timing";
+import { useService } from "@web/core/utils/hooks";
+
+const { useState, useEffect } = owl;
+
+export class IbanWidget extends CharField {
+    setup() {
+        super.setup();
+        this.state = useState({
+            isValidIBAN: null,
+        });
+        this.orm = useService("orm");
+        this.validateIbanDebounced = useDebounced(async (ev) => {
+            const iban = ev.target.value;
+            await this.validateIban(iban);
+        }, 400);
+        // initiate isValidIBAN when edition starts
+        useEffect(() => {
+            this.validateIban(this.props.value);
+        }, () => [this.props.readonly]);
+    }
+
+    onInputValidateIban(ev) {
+        this.state.isValidIBAN = null;
+        this.validateIbanDebounced(ev);
+    }
+
+    async validateIban(iban) {
+        if (!iban) {
+            this.state.isValidIBAN = null;
```

Comment:
the old version marked this case as valid. why do we have a null option?

Reply:
Yes but it was weird: the variable was set to `true` but the validation icon removed.
As in all, I make the validation icon display depends directly on this, I set it to null when the validation icon "doesn't apply".

https://github.com/odoo/odoo/blob/d9057e332d04e676a926fcc7852008349556c4a4/addons/base_iban/static/src/js/iban_widget.js#L43